assets/javascripts/discourse/components/chat-message-collapser.js

```diff
@@ -1,78 +1,125 @@
 import Component from "@ember/component";
-import { action, computed } from "@ember/object";
+import { computed } from "@ember/object";
+import { htmlSafe } from "@ember/template";
 import escape from "discourse-common/lib/escape";
 import domFromString from "discourse-common/lib/dom-from-string";
 import I18n from "I18n";
 
 export default Component.extend({
+  tagName: "",
+
   collapsed: false,
   uploads: null,
   cooked: null,
-  message: null,
 
-  @computed("uploads")
-  get imageFilename() {
-    if (this.uploads) {
-      if (this.uploads.length === 1) {
-        return this.uploads[0].original_filename;
+  @computed("cooked")
+  get youtubeCooked() {
+    const elements = Array.prototype.slice.call(domFromString(this.cooked));
+
+    return elements.reduce((acc, e) => {
+      if (youtubePredicate(e)) {
+        const id = e.dataset.youtubeId;
+        const link = `https://www.youtube.com/watch?v=${escape(id)}`;
+        const title = e.dataset.youtubeTitle;
+        const header = htmlSafe(
+          `<a target="_blank" class="chat-message-collapser-link" rel="noopener noreferrer" href="${link}">${title}</a>`
+        );
+        acc.push({ header, body: e, needsCollapser: true });
       } else {
-        return I18n.t(`chat.uploaded_files`, { count: this.uploads.length });
+        acc.push({ body: e, needsCollapser: false });
       }
-    }
+      return acc;
+    }, []);
   },
 
-  @computed("cooked")
-  get youtubeTitle() {
-    return domFromString(this.cooked).dataset.youtubeTitle;
+  @computed("uploads")
+  get uploadsHeader() {
+    if (this.uploads.length === 1) {
+      return this.uploads[0].original_filename;
+    } else {
+      return I18n.t(`chat.uploaded_files`, { count: this.uploads.length });
+    }
   },
 
   @computed("cooked")
-  get youtubeLink() {
-    const id = domFromString(this.cooked).dataset.youtubeId;
-    return `https://www.youtube.com/watch?v=${escape(id)}`;
-  },
-
-  @action
-  open() {
-    this.set("collapsed", false);
-  },
+  get imageOneboxCooked() {
+    const elements = Array.prototype.slice.call(domFromString(this.cooked));
 
-  @action
-  close() {
-    this.set("collapsed", true);
+    return elements.reduce((acc, e) => {
+      if (imageOneboxPredicate(e)) {
+        const link = animatedImagePredicate(e)
+          ? e.firstChild.src
+          : e.firstElementChild.href;
+        const header = htmlSafe(
+          `<a target="_blank" class="chat-message-collapser-link-small" rel="noopener noreferrer" href="${link}">${link}</a>`
```

Comment:
I don't think you need both `chat-message-collapser-link` (for Youtube) and `chat-message-collapser-link-small`. IMO, pick one size and go with it for all collapsable elements.

Reply:
Yeah, fair enough. Although, for expanded youtube, we now show the title twice: 

<img width="524" alt="image" src="https://user-images.githubusercontent.com/368961/149160187-f5a34b3f-d647-4315-bf39-e4e8aaf6111b.png">

Reply:
When I first made the change https://github.com/discourse/discourse-chat/pull/416 here, I actually started working on https://github.com/discourse/discourse/blob/main/plugins/lazy-yt/assets/javascripts/lib/lazyYT.js first, and added an property to hide/show the title. But I decided against changing it until someone complains about the duplicate. You're the first ;)

Reply:
Cool, 2 more complainers to go then!